gdk/commands/component/recipe_generator/BuildRecipeGenerator.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import shutil
+from pathlib import Path
+from gdk.common.CaseInsensitive import CaseInsensitiveRecipeFile, CaseInsensitiveDict
+
+import gdk.commands.component.project_utils as project_utils
+import gdk.common.consts as consts
+import gdk.common.utils as utils
+
+
+class BuildRecipeGenerator:
+    def __init__(self, project_config) -> None:
+        self.project_config = project_config
+
+    def generate(self, build_folders):
+        component_recipe = CaseInsensitiveRecipeFile().read(self.project_config["component_recipe_file"])
```

Comment:
Seems like what we are calling generators is continuously reading and writing values to the same file, ideally we read once, make the all the transformations and then write. 

Read recipe template -> transformation pipeline -> Recipe generated -> Write recipe file

Reply:
`generate` is only used once. It does what you mentioned -  reads, updates everything that is needed and then write once.